src/Component/Filter/FilterTree/FilterTree.tsx

```diff
@@ -0,0 +1,567 @@
+        break;
+      case 'or':
+        addedFilter = ['||', ['==', '', ''], ['==', '', '']];
+        break;
+      case 'not':
+        addedFilter = ['!', ['==', '', '']];
+        break;
+      case 'comparison':
+      default:
+        addedFilter = ['==', '', ''];
+        break;
+    }
+
+    if (position === '') {
+      newFilter.push(addedFilter);
+    } else {
+      const previousFilter = _get(newFilter, position);
+      previousFilter.push(addedFilter);
+      _set(newFilter, position, previousFilter);
+    }
+
+    onFilterChange(newFilter);
+  }
+
+  /**
+   * Changes a filter at a position to a given typ.
+   *
+   */
+  changeFilter = (position: string, type: string) => {
+    const {
+      filter,
+      onFilterChange
+    } = this.props;
+
+    let addedFilter: GsFilter ;
+    const newFilter: GsFilter = [...filter];
+    const previousFilter = position === '' ? newFilter : _get(newFilter, position);
+
+    switch (type) {
+      case 'and':
+        if (previousFilter && previousFilter[0] === '&&' || previousFilter[0] === '||' ) {
+          addedFilter = previousFilter;
+          addedFilter[0] = '&&';
+        } else {
+          addedFilter = ['&&', ['==', '', ''], ['==', '', '']];
+        }
+        break;
+      case 'or':
+        if (previousFilter && previousFilter[0] === '&&' || previousFilter[0] === '||' ) {
+          addedFilter = previousFilter;
+          addedFilter[0] = '||';
+        } else {
+          addedFilter = ['||', ['==', '', ''], ['==', '', '']];
+        }
+        break;
+      case 'not':
+        addedFilter = ['!', ['==', '', '']];
+        break;
+      case 'comparison':
+      default:
+        addedFilter = ['==', '', ''];
+        break;
+    }
+
+    if (position === '') {
+      onFilterChange(addedFilter);
+      return;
+    } else {
+      _set(newFilter, position, addedFilter);
+    }
+
+    onFilterChange(newFilter);
+  }
+
+  /**
+   * Removes a filter at a given position.
+   *
+   */
+  removeFilter = (position: string) => {
+    const {
+      filter,
+      onFilterChange
+    } = this.props;
+
+    const parentPosition = position.substring(0, position.length - 3);
+    const parentFilter = this.getFilterAtPosition(parentPosition);
+    let newFilter;
+
+    if (parentFilter.length <= 2) {
+      newFilter = this.removeAtPosition(filter, parentPosition);
+    } else {
+      newFilter = this.removeAtPosition(filter, position);
+    }
+
+    onFilterChange(newFilter);
+  }
+
+  /**
+   * Transforms a position String like '[2][3]' to an positionArray like [2, 3].
+   */
+  positionStringAsArray = (positionString: string) => {
+    return positionString
+      .replace('][', ',')
+      .replace(']', '')
+      .replace('[', '')
+      .split(',')
+      .map(i => parseInt(i, 10));
+  }
+
+  /**
+   * Transforms am positionArray like [2, 3] to a string like '[2][3]'.
+   */
+  positionArrayAsString = (positionArray: number[]) => {
```

Comment:
```suggestion
  positionArrayAsString = (positionArray: [number, number]) => {
```
Through this change, positionArray is a tuple of two numbers instead of an arbitrary array. This will make sure, you only have an array of two numbers. Any other structure will fail. Also useful, because .replace() only replaces the first occurence of the string if not used with a regexp with global flag.

Reply:
I fixed the replace method in 6b1f652